js/tests/unit/carousel.spec.js

```diff
@@ -1056,13 +1056,29 @@ describe('Carousel', () => {
       ].join('')
 
       const carouselEl = fixtureEl.querySelector('#myCarousel')
+      const addEventSpy = spyOn(carouselEl, 'addEventListener').and.callThrough()
+      const removeEventSpy = spyOn(carouselEl, 'removeEventListener').and.callThrough()
+
       const carousel = new Carousel(carouselEl)
 
-      spyOn(EventHandler, 'off').and.callThrough()
+      let expectedArgs = [
+        ['keydown', jasmine.any(Function), jasmine.any(Boolean)],
+        ['mouseover', jasmine.any(Function), jasmine.any(Boolean)],
+        ['mouseout', jasmine.any(Function), jasmine.any(Boolean)]
+      ]
+
+      if (carousel._touchSupported) {
+        expectedArgs = expectedArgs.concat([
+          ['pointerdown', jasmine.any(Function), jasmine.any(Boolean)],
+          ['pointerup', jasmine.any(Function), jasmine.any(Boolean)]
+        ])
+      }
```

Comment:
@rohit2sharma95 

One strange thing I noticed is that the Node 12/14 tests were failing for carousel.  It seems `_touchSupported` is evaluating to false only in those two environments.  Node 10 was ok.

Reply:
It seems CI is green there, so I guess we are all good?

Reply:
Yeah it's good to go @XhmikosR.  Had to put if the `if (carousel._touchSupported)` to pass the CI tests.  Thought it was worth mentioning since it caught me by surprise :)

Reply:
Hmm, in that can we might want to look at this closely. The check shouldn't fail and if it does it means we might be doing it wrong. I recall that this check is a little hackish.

@patrickhlauke: the offending line is https://github.com/twbs/bootstrap/blob/868c8b8bce363849f495ea5fcd13ffbee4bf82a7/js/src/carousel.js#L120

Reply:
Here's the link to the failed test run:  https://github.com/twbs/bootstrap/runs/1846236124

Reply:
So, I have no idea why tests fail. Can you please add a comment why you added the `carousel._touchSupported` check so that it's documented?

Reply:
don't quite understand what IS actually failing in those test runs, but ... maybe whatever is running the tests has no concept of / has not enabled pointer events? and those lines are failing because of that? and it also doesn't have touch events enabled, so evaluated `_touchSupported` to false and so skips the lines that would cause it to blow up?

Reply:
actually, if i understand this right...this tests if those specific events have been removed or something? if that's the case, it's probably blowing up because the pointer events were not *added* in the first place since the environment likely doesn't signal that it supports touch. so it's right that if you then indiscriminately check if pointer events were *removed* if would fail since they weren't added to begin with?

Reply:
@XhmikosR, I've added a comment into the test.

@patrickhlauke, actually it blows up in both add/remove assertions.  This test ensures that the correct number of event handlers are added and removed.  I was expecting 5 events to be add/removed here since the default config value for `touch` is `true`.  

I looked into other touch tests to see how this is passing and it looks like this line is appended prior to the test:
`document.documentElement.ontouchstart = () => {}`.  I suppose this is already a known issue then?  I could prefix the dispose test with that same line and remove the `_touchSupported` check.

Reply:
thanks for clarifying. well, the headless browser will report that it doesn't support touch, so to test that aspect yes, that line patches in / fakes the necessary property/function to then make it look like the test environment does support touch.

in which case yes, this should also be patched in before the other test for removal runs (before the carousel is even instantiated, so it adds those pointer events because it thinks the environment supports touch), so that you can THEN test that they're removed correctly.

Reply:
and probably worth adding a comment to it here (and, if not already present, on that line for the other test) to clarify this is done to force/fake touch support which the test environment/headless browser otherwise doesn't have

Reply:
maybe even add an extra line here and in the other place to also set `navigator.maxTouchPoints` to a value greater than zero, if the test environment runner allows this sort of override of readonly values

Reply:
Thanks @patrickhlauke - made the changes. The headless browser doesn't allow setting `maxTouchPoints`, so I left that alone.